libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -341,6 +341,7 @@ void RangeFinder::detect_instance(uint8_t instance, uint8_t& serial_instance)
 {
     const Type _type = (Type)params[instance].type.get();
     switch (_type) {
+#if AP_RANGEFINDER_ENABLE
```

Comment:
If we aren't going to define out the whole file, then I would move this to define out the whole function instead.  You are not getting a driver anyways so who cares about the parameters coming from the non-existent backends.... 

And move the matching #endif to the end of the function.

This saves approx. 800B

```suggestion
{
#if AP_RANGEFINDER_ENABLE
    const Type _type = (Type)params[instance].type.get();
    switch (_type) {
```